pkgs/servers/onlyoffice-documentserver/default.nix

```
@@ -12,27 +12,27 @@
 }:
 
 let
-  version = "7.4.1";
-  tag = lib.concatStringsSep "." (lib.take 3 (lib.splitVersion version));
-  dist = {
-    x86_64-linux = rec {
-      archSuffix = "amd64";
-      url = "https://github.com/ONLYOFFICE/DocumentServer/releases/download/v${tag}/onlyoffice-documentserver_${archSuffix}.deb";
-      hash = "sha256-60S8M1Y9BxuMxXGxEaxW82Va5lSnZZPfQnPq2ivTXdU=";
+  lookup = {
+    x86_64-linux = {
+      fileSuffix = "amd64.deb";
+      fileHash = "sha256-60S8M1Y9BxuMxXGxEaxW82Va5lSnZZPfQnPq2ivTXdU=";
     };
-    aarch64-linux = rec {
-      archSuffix = "arm64";
-      url = "https://github.com/ONLYOFFICE/DocumentServer/releases/download/v${tag}/onlyoffice-documentserver_${archSuffix}.deb";
-      hash = "sha256-z2Mp9q+KLpQLG3tVUZaD3jMYoRAXoq7s5wUhNnckTDg=";
+    aarch64-linux = {
+      fileSuffix = "arm64.deb";
+      fileHash = "sha256-z2Mp9q+KLpQLG3tVUZaD3jMYoRAXoq7s5wUhNnckTDg=";
     };
   };
   # var/www/onlyoffice/documentserver/server/DocService/docservice
-  onlyoffice-documentserver = stdenv.mkDerivation {
+  onlyoffice-documentserver = stdenv.mkDerivation rec {
     pname = "onlyoffice-documentserver";
-    inherit version;
-    src = fetchurl {
-      inherit (dist.${stdenv.hostPlatform.system} or
-      (throw "Unsupported system: ${stdenv.hostPlatform.system}")) url hash;
+    version = "7.4.1";
+    src = let
+      inherit (lookup.${stdenv.hostPlatform.system} or (throw "Unsupported system: ${stdenv.hostPlatform.system}")) fileSuffix fileHash;
+      tag = lib.concatStringsSep "." (lib.take 3 (lib.splitVersion version));
+      url = "https://github.com/ONLYOFFICE/DocumentServer/releases/download/v${tag}/onlyoffice-documentserver_${fileSuffix}";
+      hash = "${fileHash}";
+    in fetchurl {
+      inherit url hash;
```

```suggestion
    src = let
      inherit (lookup.${stdenv.hostPlatform.system} or (throw "Unsupported system: ${stdenv.hostPlatform.system}")) arch hash;
    in fetchurl {
      "https://github.com/ONLYOFFICE/DocumentServer/releases/download/v${version}/onlyoffice-documentserver_${arch}.deb";
      inherit hash;
```
no idea why we did this before but it seems no longer necessary, so we can just simplify the code a lot